Client/Frontend/Theme/ThemeManager.swift

```diff
@@ -9,101 +9,72 @@ import Shared
 /// through the `themeManager` variable, while using dependency injection.
 protocol Themeable { }
 
-extension Themeable {
-    var themeSystem: ThemeManager { return ThemeManager.shared }
+extension Themeable {}
+
+enum ThemeType: String {
+    case light = "normal" // This needs to match the string used in the legacy system
+    case dark
+
+    func getInterfaceStyle() -> UIUserInterfaceStyle {
+        switch self {
+        case .light:
+            return .light
+        case .dark:
+            return .dark
+        }
+    }
+}
+
+protocol ThemeManager {
+    var currentTheme: Theme { get }
+
+    func getInterfaceStyle() -> UIUserInterfaceStyle
 }
 
 /// The `ThemeManager` will be responsible for providing the theme throughout the app
-///
-/// IMPORTANT NOTE:
-/// The current functionality of `ThemeManager` is restricted to providing the default
-/// theme. The rest of the functionality, is a sketching out of how implementing user
-/// defined themes might work. It's a work in progress and needs not be touched until
-/// such a feature is actually added.
-final class ThemeManager {
+final class DefaultThemeManager: ThemeManager {
```

Comment:
Why default and not just "ThemeManager" ?

Reply:
The protocol is the thing that will be referenced the most so I wanted it's name to be ThemeManager. I could have also called the class ThemeManagerImplementation but I wanted to leave the door open to there potentially being a second theme manager in the future. Like if Roux's dream comes true and we have user configurable themes at some point, the best way to handle that would probably be to have a second CustomThemeManager that handles all the logic of saving/loading the entire theme to/from disk etc.

Reply:
MY DREAMS OF SKITTLE VOMIT FIREFOX!